packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+          return this.account;
+        }
+        default:
+          this.account = null;
+          console.log('The network has not been set.');
+      }
+      return this.account;
+    }
+    if (arguments.length === 1 && typeof (arguments[0]) === 'string') {
+      try {
+        switch (this.network) {
+          case 'stellar': {
+            if (arguments[0].charAt(0) === 'S') {
+              // generate account from private key
+              this.account = new Account(this.network);
+              await this.account.setAccount(StellarSdk.Keypair.fromSecret(arguments[0]));
+              this.accountArray.push(Object.assign({}, this.account));
+            } else {
+              this.account = null;
+              console.log('The starting char must be S (private key)');
+            }
+            break;
+          }
+          case 'ethereum': {
+            this.account = new Account(this.network);
+            await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(arguments[0]));
+            this.accountArray.push(Object.assign({}, this.account));
+            break;
+          }
+          default:
+            this.account = null;
+            console.log('The network has not been set.');
+        }
+        return null;
+      } catch (err) {
+        console.log('The input is not a valid private/public key.');
+        return null;
+      }
+    } else {
+      console.log('The argument must be empty, or 0,1,2,...');
+      return null;
+    }
+  }
+
+  /**
+   * A shorthand for creating multiple accounts.
+   * The accounts created will be the 0th, 1st, ... (n-1)th in the wallet
+   * @param {int} numberOfAccounts
+   */
+  async setMultipleAccounts(numberOfAccounts) {
+    for (let i = 0; i < numberOfAccounts; i++) {
+      await this.getAccount(i);
+    }
+  }
+
+  /**
+   * Set the current account.
+   * @param {number} index - The index of the account in the accountArray
+   */
+  setCurrentAccount(index) {
+    const max = this.accountArray.length - 1;
+    if (index > max) {
+      console.log(`the maximum index is${max.toString()}`);
+    } else {
+      this.account = this.accountArray[index];
+    }
+  }
+
+  /**
+   * Return the account array.
+   * The index of accounts in this array can be different from the index
+   * of the account in the wallet,
+   * depending on the sequence the accounts are created.
+   * @returns {Array} the array of accounts
+   */
+  getAccountArray() {
+    if (this.accountArray == null) {
+      console.log('The account array is empty');
+      return null;
+    }
+    return this.accountArray;
+  }
+
+  /**
+   *
+   * @param {string} password - the password
+   * @returns {ByteStringBuffer} the object containing the encrypted string
+   */
+  encryptSeed(password) {
+    // generate a random key and IV
+    // Note: a key size of 16 bytes will use AES-128, 24 => AES-192, 32 => AES-256
+    this.iv = forge.random.getBytesSync(16);
```

Comment:
Will be awesome to mention how forge seeds its prng. Quite important for generating randomness

https://www.npmjs.com/package/node-forge#prng

Reply:
added to comments